emspy/query/analyticset.py

```diff
@@ -23,109 +28,150 @@ def __init__(self, conn, ems_id, exclude_trees=[], exclude_patterns=[]):
+
+            path_list = self._analytic_set_path.split('\\')
+            if not path_only:
+                self._analytic_set_id = path_list.pop()
+                # In case the path ended with "\"
+                if self._analytic_set_id == '':
+                    raise ValueError('Missing analytic set name. It should be at the end of the path')
+            
+            # Remove empty items from list
+            path_list = [i for i in path_list if i]
+
+            if (path_list[0].lower() == 'parameter sets') or (path_list[0].lower() == 'root'):
+                path_list.pop(0)
+
+            if len(path_list) == 0:
+                self._group_id = 'root'
+            else:
+                # The group_id is just the folder path separated by ":"
+                self._group_id = ":".join(path_list)
+
+    def get_analytic_set(self, analytic_set_path):
         """
-        Update analytic sets list
+        Gets analytic the data for the analytic set in the analytic_set_path
+
+        Parameters
+        ----------
+        analytic_set_path: str
+            The path (from root using "\" as separator) to the analytic set
 
         Returns
         -------
-        None
+        DataFrame
         """
-        self.__get_analytic_set_group('Root')
-        self._analytic_sets = pd.DataFrame(self._analytic_sets)
-        self._analytic_sets = self._analytic_sets.drop(['groups', 'sets'], axis=1)\
-            .join(self._analytic_sets['sets'].apply(pd.Series))\
-            .melt(id_vars=['name', 'groupId'], value_name='set')\
-            .drop('variable', axis=1)\
-            .dropna(subset=['set'])
-        self._analytic_sets['set'] = self._analytic_sets['set'].apply(lambda d: d['name'])
-        self._analytic_sets.sort_values(['name', 'groupId', 'set'], inplace=True)
-        self._analytic_sets.reset_index(drop=True, inplace=True)
-
-    def select_set(self, analytics_set_name=None, searchtype='match'):
+        if not analytic_set_path:
+            raise ValueError("No Analytic Set path was passed in")
+        self._analytic_set_path = analytic_set_path
+        self.__parse_path()
+        
+        print(f'-- Fetching analytic set "{self._analytic_set_path}"')
+        try:
+            _, dict_data = self._conn.request(
+                uri_keys=('analyticSet', 'analytic_set'),
+                uri_args=(self._ems_id, self._group_id, self._analytic_set_id)
+            )
+
+            flat_analytic_set_dict = self.__parse_analytic_set(dict_data)
+            self._analytic_set_name = flat_analytic_set_dict['name']
+            self._analytic_set_description = flat_analytic_set_dict['description']
+            data_df = pd.DataFrame.from_records(flat_analytic_set_dict['items'])
+
+            analytic_set_df = pd.DataFrame(columns=AnalyticSet.__analytic_set_columns)
+
+            analytic_set_df = pd.concat([data_df,analytic_set_df])
```

Comment:
Just a minor style thing but I would add a space after the comma in the list since we've been doing that other places

Reply:
done